src/yunohost/user.py

```diff
@@ -462,6 +469,276 @@ def user_info(auth, username):
+    conflict = auth.get_conflict({
+        'cn': groupname
+    }, base_dn='ou=groups,dc=yunohost,dc=org')
+    if conflict:
+        raise YunohostError('group_name_already_exist', name=groupname)
+
+    # Validate uniqueness of groupname in system group
+    all_existing_groupnames = {x.gr_name for x in grp.getgrall()}
+    if groupname in all_existing_groupnames:
+        raise YunohostError('system_groupname_exists')
+
+    if not gid:
+        # Get random GID
+        all_gid = {x.gr_gid for x in grp.getgrall()}
+
+        uid_guid_found = False
+        while not uid_guid_found:
+            gid = str(random.randint(200, 99999))
+            uid_guid_found = gid not in all_gid
+
+    attr_dict = {
+        'objectClass': ['top', 'groupOfNamesYnh', 'posixGroup'],
+        'cn': groupname,
+        'gidNumber': gid,
+    }
+
+    if auth.add('cn=%s,ou=groups' % groupname, attr_dict):
+        logger.success(m18n.n('group_created'))
+        if sync_perm:
+            permission_sync_to_user(auth)
+        return {'name': groupname}
+
+    raise YunohostError('group_creation_failed')
+
+
+@is_unit_operation([('groupname', 'user')])
+def user_group_delete(operation_logger, auth, groupname, force=False, sync_perm=True):
+    """
+    Delete user
+
+    Keyword argument:
+        groupname -- Groupname to delete
+
+    """
+    from yunohost.permission import permission_sync_to_user
+
+    if not force and (groupname == 'all_users' or groupname == 'admins' or groupname in user_list(auth, ['uid'])['users']):
+        raise YunohostError('group_deletion_not_allowed', user=groupname)
+
+    operation_logger.start()
+    if not auth.remove('cn=%s,ou=groups' % groupname):
+        raise YunohostError('group_deletion_failed')
+
+    logger.success(m18n.n('group_deleted'))
+    if sync_perm:
+        permission_sync_to_user(auth)
+
+
+@is_unit_operation([('groupname', 'user')])
+def user_group_update(operation_logger, auth, groupname, add_user=None, remove_user=None, force=False, sync_perm=True):
+    """
+    Update user informations
+
+    Keyword argument:
+        groupname -- Groupname to update
+        add_user -- User to add in group
+        remove_user -- User to remove in group
+
+    """
+
+    from yunohost.permission import permission_sync_to_user
+
+    attrs_to_fetch = ['member']
+
+    if (groupname == 'all_users' or groupname == 'admins') and not force:
+        raise YunohostError('edit_group_not_allowed', group=groupname)
+
+    # Populate group informations
+    result = auth.search(base='ou=groups,dc=yunohost,dc=org',
+                         filter='cn=' + groupname, attrs=attrs_to_fetch)
+    if not result:
+        raise YunohostError('group_unknown', group=groupname)
+    group = result[0]
+
+    new_group_list = {'member': set(), 'memberUid': set()}
+    if 'member' in group:
+        new_group_list['member'] = set(group['member'])
+    else:
+        group['member'] = []
+
+    user_l = user_list(auth, ['uid'])['users']
```

Comment:
Eh, when possible please let's explicit the arguments to improve readability :S 

So in that case : `user_list(auth, fields=['uid'])` instead of just `user_list(auth, ['uid'])`